sdk/src/keypairs.rs

```diff
@@ -0,0 +1,113 @@
+use crate::pubkey::Pubkey;
+use crate::signature::{KeypairUtil, Signature};
+
+pub trait Keypairs {
+    fn pubkeys(&self) -> Vec<Pubkey>;
+    fn sign_message(&self, message: &[u8]) -> Vec<Signature>;
+}
+
+impl<T: KeypairUtil> Keypairs for [&T] {
+    fn pubkeys(&self) -> Vec<Pubkey> {
+        self.iter().map(|keypair| keypair.pubkey()).collect()
+    }
+
+    fn sign_message(&self, message: &[u8]) -> Vec<Signature> {
+        self.iter()
+            .map(|keypair| keypair.sign_message(message))
+            .collect()
+    }
+}
+
+impl<T: KeypairUtil, U: KeypairUtil> Keypairs for (T, U) {
+    fn pubkeys(&self) -> Vec<Pubkey> {
+        vec![self.0.pubkey(), self.1.pubkey()]
+    }
+
+    fn sign_message(&self, message: &[u8]) -> Vec<Signature> {
+        vec![self.0.sign_message(message), self.1.sign_message(message)]
+    }
+}
+
+impl<T: KeypairUtil, U: KeypairUtil, V: KeypairUtil> Keypairs for (T, U, V) {
```

Comment:
macros work well for this

Reply:
Wrote a terrible one to dedup the code, thanks.  Unfortunately, no time to make a macro to invoke that macro ~10 times.